kernel/dev/serial/rs232.cpp

```diff
@@ -125,33 +129,11 @@ void rs232_init(uint16_t com_id) {
     );
 }
 
-ring_buff_t* rs232_init_buffer(int size) {
-    mutex_lock(&mutex_rs232);
-    // Allocate space for the input buffer
-    read_buffer = (ring_buff_t*)malloc(sizeof(ring_buff_t));
-    // Initialize the ring buffer
-    if (ring_buffer_init(read_buffer, size) == 0) {
-        mutex_unlock(&mutex_rs232);
-        return read_buffer;
-    } else {
-        mutex_unlock(&mutex_rs232);
-        return NULL;
-    }
-}
-
-ring_buff_t* rs232_get_buffer() {
-    // Can return NULL. This is documented.
-    return read_buffer;
-}
-
 char rs232_get_char() {
     mutex_lock(&mutex_rs232);
     // Grab the last byte and convert to a char
     uint8_t data = 0;
-    if (read_buffer != NULL)
-    {
-        ring_buffer_dequeue(read_buffer, &data);
-    }
+    ring.Dequeue(&data);
```

Comment:
Do we need to check the return code here?

Reply:
The worst thing that happens is a null character gets returned, and I'm not sure what else we would do besides that, so I'm not sure it's really worth checking the return value. What would we do in event of failure besides return 0 / null character?